### ISSUE TITLE
Testing with desktop captured image, I got a blur output.

### ISSUE THREAD

**Issue by steven8274** [pkuxmq/Invertible-Image-Rescaling#39]
Firstly, many thanks for the authors' great job on image rescaling!I've trained the model in 'IRN 4x mode', and got a really good result with 'DIV2K' validation dataset.But when I use a downscaled screenshot image as the HR image, the output SR image is not good enough.Firstly, I captured my desktop into an image, which size is 3840x2160, then downscale it with 'ffmpeg' to 1920x1080, and then use this 1920x1080 image as the test image, I got an blur SR image.The characters in this SR image are all blurred, which are much more diffcult to recognize than those in ground truth 1920x1080 image.But if I use the original 3840x2160 image to test, the characters will not  be blurred.

**The original 3840x2160 image(ground truth):**
[3840x2160-gt](https://photos.google.com/photo/AF1QipN-GhQrxvIPLmjM86j8kB-lLlaeAtt8Z8ocyRgH)
**The SR of Raw 3840x2160 image:**
[3840x2160-sr](https://photos.google.com/album/AF1QipN52LfFzBEp-z_IgF2lenlBaIdzb_SGYzeEI17B/photo/AF1QipMTqPxbbozlbkrxFS3gyhiuD3UIsZoPGnLUEV44)
**The donwscaled 1920x1080 image(ground truth):**
[1920x1080-gt](https://photos.google.com/album/AF1QipN52LfFzBEp-z_IgF2lenlBaIdzb_SGYzeEI17B/photo/AF1QipMxJh5geCVA0x4dlGZUxmj22TlZhrlmqCIoOyJw)
**The SR of donwscaled 1920x1080 image:**
[1920x1080-sr](https://photos.google.com/album/AF1QipN52LfFzBEp-z_IgF2lenlBaIdzb_SGYzeEI17B/photo/AF1QipPRY90WeqZbvUxx74ULoWfb3hmOFwtCyLeRModH)

I trained the model with 'DIV2K' dataset, I found that all images in this dataset are very clear ones.So, should test images also be very clear ones?Is the 1080p(1920x1080) image not clear enough?

**Comment by steven8274** [comment]
I found that this problem only occured when input image's resolution is under 2K, e.g. 1920x1080, 1440x900, etc.

**Comment by steven8274** [comment]
I alse found that IRN_x2 model does not cause such problem.Besides, IRN+_4x model cause such problem too.

**Comment by pkuxmq** [comment]
Probably because the model is only trained on high-resolution images (DIV2K) and does not generalize very well for some unseen distributions of input images. For real applications, you may slightly finetune the model for your data distribution, e.g. capture several downscaled images to form a tiny dataset and finetune the model for several epochs (or you may finetune with downscaled DIV2K images as inputs, which should encourage the generalization to such data distributions).

**Comment by steven8274** [comment]
> Probably because the model is only trained on high-resolution images (DIV2K) and does not generalize very well for some unseen distributions of input images. For real applications, you may slightly finetune the model for your data distribution, e.g. capture several downscaled images to form a tiny dataset and finetune the model for several epochs (or you may finetune with downscaled DIV2K images as inputs, which should encourage the generalization to such data distributions).

Thanks a lot！I'll try that.
I also found another problem with IRN_x4 and IRN+_x4 model:
https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/13#issuecomment-1092393645
The downscaled image is added with some color boxes which don't exist in ground truth image.This problem only occur when I use the 'IRN_x4' model I trained.When the pretrained models(IRN_x2、IRN_x4 adn IRN+_x4) you provided, I haven't met this problem yet.But I worry about that.

**Comment by pkuxmq** [comment]
The artifacts are likely to be caused by unrobustness of LR generation for some unseen distributions. One solution is also to finetune models on several samples of the target distribution. For example, if we finetune the model for several epochs on Kodak, the artifacts on Kodak with pretrained models can be removed. Another method is to increase the loss weight for LR guidance during training. This may lead to a slight performance drop for HR reconstruction (around 0.1-0.2 dB if we increase the weight by 10 times), but the restriction for valid LR image is stronger. Also, there may be other methods to regularize the model from overfitting that require exploration.

**Comment by steven8274** [comment]
> The artifacts are likely to be caused by unrobustness of LR generation for some unseen distributions. One solution is also to finetune models on several samples of the target distribution. For example, if we finetune the model for several epochs on Kodak, the artifacts on Kodak with pretrained models can be removed. **Another method is to increase the loss weight for LR guidance during training.** This may lead to a slight performance drop for HR reconstruction (around 0.1-0.2 dB if we increase the weight by 10 times), but the restriction for valid LR image is stronger. Also, there may be other methods to regularize the model from overfitting that require exploration.

Thanks!Could you please tell me how to do this(bold part) in codes?I havn't understand your model and codes very well in details.

**Comment by pkuxmq** [comment]
In config files, e.g. options/train/train_IRN_x4.yml, modify the variable "lambda_fit_forw", e.g. from 16 to 160.

**Comment by steven8274** [comment]
> In config files, e.g. options/train/train_IRN_x4.yml, modify the variable "lambda_fit_forw", e.g. from 16 to 160.

Much appreciated!

**Comment by steven8274** [comment]
I've continually train the pretrained model 'IRN_x2.pth' with another 2500 epochs with kodak image set.As a result, the kodak images' artifacts are gone, but the DIV2k images have artifacts however.
I'll try to increase the loss weight for LR guidance(from 16 to 160), and report the result later.

**Comment by steven8274** [comment]
After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.

**Comment by mrgreen3325** [comment]
> After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.

Hello
May I know which optimal "lambda_fit_forw" do you choose?
Is that you want to obtain the LR for other purposes?
For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.

**Comment by steven8274** [comment]
> > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> 
> Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.

I should have set "lambda_fit_forw" to 160.
https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661

**Comment by mrgreen3325** [comment]
> > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > 
> > 
> > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> 
> I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)

Thanks for reply.
I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40).
Is the LR nice for  "lambda_fit_forw" to 160?
Is there any other thing I need to adjust to obtain a nice LR?

**Comment by steven8274** [comment]
> > > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > > 
> > > 
> > > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> > 
> > 
> > I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)
> 
> Thanks for reply. I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40). Is the LR nice for "lambda_fit_forw" to 160? Is there any other thing I need to adjust to obtain a nice LR?

For 'IRN_x2' I set set it to 40, for 'IRN_x4' I set it to 160.

**Comment by mrgreen3325** [comment]
> > > > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > > > 
> > > > 
> > > > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> > > 
> > > 
> > > I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)
> > 
> > 
> > Thanks for reply. I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40). Is the LR nice for "lambda_fit_forw" to 160? Is there any other thing I need to adjust to obtain a nice LR?
> 
> For 'IRN_x2' I set set it to 40, for 'IRN_x4' I set it to 160.

Thanks for reply.
May I know how many epoch or iteration you train to avoid the artifact?

**Comment by steven8274** [comment]
> > > > > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > > > > 
> > > > > 
> > > > > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> > > > 
> > > > 
> > > > I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)
> > > 
> > > 
> > > Thanks for reply. I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40). Is the LR nice for "lambda_fit_forw" to 160? Is there any other thing I need to adjust to obtain a nice LR?
> > 
> > 
> > For 'IRN_x2' I set set it to 40, for 'IRN_x4' I set it to 160.
> 
> Thanks for reply. May I know how many epoch or iteration you train to avoid the artifact?

I trained the pretrained model the author suplied with another 5000 epochs.

**Comment by mrgreen3325** [comment]
5000 epochs, it takes 2 weeks to do that?

**Comment by mrgreen3325** [comment]
> > > > > > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > > > > > 
> > > > > > 
> > > > > > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> > > > > 
> > > > > 
> > > > > I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)
> > > > 
> > > > 
> > > > Thanks for reply. I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40). Is the LR nice for "lambda_fit_forw" to 160? Is there any other thing I need to adjust to obtain a nice LR?
> > > 
> > > 
> > > For 'IRN_x2' I set set it to 40, for 'IRN_x4' I set it to 160.
> > 
> > 
> > Thanks for reply. May I know how many epoch or iteration you train to avoid the artifact?
> 
> I trained the pretrained model the author suplied with another 5000 epochs.

How long did it take to train? Is 5000 epochs close to 2 millions iterations?

**Comment by steven8274** [comment]
> > > > > > > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > > > > > > 
> > > > > > > 
> > > > > > > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> > > > > > 
> > > > > > 
> > > > > > I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)
> > > > > 
> > > > > 
> > > > > Thanks for reply. I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40). Is the LR nice for "lambda_fit_forw" to 160? Is there any other thing I need to adjust to obtain a nice LR?
> > > > 
> > > > 
> > > > For 'IRN_x2' I set set it to 40, for 'IRN_x4' I set it to 160.
> > > 
> > > 
> > > Thanks for reply. May I know how many epoch or iteration you train to avoid the artifact?
> > 
> > 
> > I trained the pretrained model the author suplied with another 5000 epochs.
> 
> How long did it take to train? Is 5000 epochs close to 2 millions iterations?

About half a day?I use a GeForce GTX 1080-Ti gpu.

**Comment by mrgreen3325** [comment]
> > > > > > > > > After I changed '**the loss weight for LR guidance(from 4 to 40)**' and continually trained the pretrained 'IRN_x2' model with 5000 epochs, the model is more robust with all these image sets.DIV2k, Kodak, Set5, Set14, 100urban and B100 all are processed well.There is no any artifacts now!Although, the reconstructed images may be more blurred by a little than before.I think this 'weight' setting is more acceptable.
> > > > > > > > 
> > > > > > > > 
> > > > > > > > Hello May I know which optimal "lambda_fit_forw" do you choose? Is that you want to obtain the LR for other purposes? For my case, I want to obtain some fancy LR other than bicubic LR for another training purpose.
> > > > > > > 
> > > > > > > 
> > > > > > > I should have set "lambda_fit_forw" to 160. [#39 (comment)](https://github.com/pkuxmq/Invertible-Image-Rescaling/issues/39#issuecomment-1094650661)
> > > > > > 
> > > > > > 
> > > > > > Thanks for reply. I thought you set it to 40 since you said the loss weight for LR guidance(from 4 to 40). Is the LR nice for "lambda_fit_forw" to 160? Is there any other thing I need to adjust to obtain a nice LR?
> > > > > 
> > > > > 
> > > > > For 'IRN_x2' I set set it to 40, for 'IRN_x4' I set it to 160.
> > > > 
> > > > 
> > > > Thanks for reply. May I know how many epoch or iteration you train to avoid the artifact?
> > > 
> > > 
> > > I trained the pretrained model the author suplied with another 5000 epochs.
> > 
> > 
> > How long did it take to train? Is 5000 epochs close to 2 millions iterations?
> 
> About half a day?I use a GeForce GTX 1080-Ti gpu.

you use a small dataset to train it?

**Comment by steven8274** [comment]
> 

Use 'DIV2k' too.

**Comment by mrgreen3325** [comment]
> > 
> 
> Use 'DIV2k' too.

you mean the niter = 5000?

**Comment by steven8274** [comment]
> > > 
> > 
> > 
> > Use 'DIV2k' too.
> 
> you mean the niter = 5000?

Yes!Sorry for misleading you.

**Comment by mrgreen3325** [comment]
> > > > 
> > > 
> > > 
> > > Use 'DIV2k' too.
> > 
> > 
> > you mean the niter = 5000?
> 
> Yes!Sorry for misleading you.

Thanks for reply.
Did you use the kodak dataset to train it as well?

**Comment by Feynman1999** [comment]
> kodak

Hello, may I ask if there is any instability in your loss during training on div2k? Also, it seems that the beginning of the training will not reach a high PSNR, but will gradually increase, right?
